pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go

```diff
@@ -150,7 +150,7 @@ func (c *SyncController) buildWorks(quota *policyv1alpha1.FederatedResourceQuota
 			errs = append(errs, err)
 			continue
 		}
-		workName := names.GenerateWorkName("ResourceQuota", quota.Name, quota.Namespace)
+		workName := names.GenerateWorkNameLegacy("ResourceQuota", quota.Name, quota.Namespace)
```

Comment:
For works created by dedicated controllers, they are still using legacy names, I think it's fine since name collisions would not happen

We can use new naming for them, that would add some boilerplate code in those controllers though